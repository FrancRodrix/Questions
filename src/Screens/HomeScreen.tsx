import { useNavigation } from "@react-navigation/core";
import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  TextInput,
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  FlatList,
} from "react-native";
import Axios from "axios";
import { padding } from "styled-system";

interface Props {
  navigation;
}

const HomeScreen = ({navigation}) => {
  const [keyWord, setKeyword] = useState("");
  const [api, setApi] = useState("https://restcountries.com/v2/name/");
  const [capital, setCapital] = useState("");

  const [countryList, setCountryList] = useState();

  const fetchApi = async (api: any) => {
    const data = api;
    try {
      const response = await Axios.get(data).then(async (res) => {
        // console.log(res.data, "FETCHED");
        setCapital(res.data);
        setCountryList(res.data);
        console.log(res.data)
      });
    } catch (e) {
      console.log(e);
    }
  };

  const handleSearch = () => {
    console.log(keyWord);
    fetchApi(api + keyWord);
  };

  const renderItem = (item: any) => {
    return (
      <TouchableOpacity
        onPress={() => {
          navigation.navigate("Details", {
            data: item,
          });
        }}
        style={styles.countryCard}
      >
        <Text style={{ textAlign: "center" }}>{item.item.name}</Text>
        <Text style={{ textAlign: "center" }}>{item.item.capital}</Text>
      </TouchableOpacity>
    );
  };
  return (
    <SafeAreaView>
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          width: "80%",
          alignSelf: "center",
          marginVertical: 20,
        }}
      >
        <View style={styles.searchBar}>
          <TextInput
            testID="Input"
            onChangeText={(text: any) => {
              setKeyword(text);
            }}
            value={keyWord}
            placeholder={"Enter Keyword"}
            placeholderTextColor={"#fff"}
          />
        </View>
        <View style={{ flex: 1 }}>
          <TouchableOpacity
            onPress={handleSearch}
            disabled={keyWord.length == 0}
            style={styles.button}
          >
            <Text style={styles.buttonText}>Search</Text>
          </TouchableOpacity>
        </View>
      </View>

      <FlatList data={countryList} 
      renderItem={renderItem}
      keyExtractor={item=>item.alpha2Code} />
    </SafeAreaView>
  );
};

export default HomeScreen;

const styles = StyleSheet.create({
  searchBar: {
    flex: 3,
    // alignSelf: 'center',
    width: "80%",
    backgroundColor: "#fe5674",
    padding: 10,
    borderRadius: 15,
    marginRight: 15,
  },
  button: {
    // alignSelf: 'center',
    padding: 10,
    backgroundColor: "#3D85C6",
    borderRadius: 15,
    // marginTop: 20,
  },
  buttonText: {
    color: "#fff",
    textAlign: "center",
  },
  countryCard: {
    alignSelf: "center",
    borderWidth: 1,
    borderColor: "#000",
    borderRadius: 15,
    padding: 20,
    marginBottom: 15,
    width: "80%",
    backgroundColor: "yellow",
  },
});
