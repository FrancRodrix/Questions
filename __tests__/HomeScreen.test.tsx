import "react-native";
import React from "react";
import HomeScreen from "../src/Screens/HomeScreen";
import renderer from "react-test-renderer";

const navigation = {
  navigate: jest.fn(),
};

test("First test case", () => {
  expect(1 + 5).toBe(6);
});

test("renders correctly", () => {
  const tree = renderer.create(<HomeScreen />);
  expect(tree).toMatchSnapshot();
});

test("States working correctly", () => {
  const tree = renderer.create(<HomeScreen navigation={navigation} />);
  const textInput = tree.root.findByProps({ testID: "Input" }).props;
  renderer.act(() => textInput.onChangeText("Test"));
  const textInput1 = tree.root.findByProps({ testID: "Input" }).props;
  expect(textInput1.value).toBe("Test");
});
