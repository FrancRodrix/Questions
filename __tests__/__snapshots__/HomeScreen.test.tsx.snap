// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "marginVertical": 20,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fe5674",
          "borderRadius": 15,
          "flex": 3,
          "marginRight": 15,
          "padding": 10,
          "width": "80%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter Keyword"
        placeholderTextColor="#fff"
        rejectResponderTermination={true}
        testID="Input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3D85C6",
            "borderRadius": 15,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "textAlign": "center",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
